app/client/ui/DocMenu.ts

```diff
@@ -529,6 +533,16 @@ export function makeDocOptionsMenu(home: HomeModel, doc: Document, renaming: Obs
       dom.cls('disabled', !roles.canEdit(orgAccess)),
       testId('pin-doc')
     ),
+    menuItem(() => {
+        const {appModel} = page;
+        if (!appModel.currentValidUser) {
+          page.clearUnsavedChanges();
+          window.location.href = getLoginOrSignupUrl({srcDocId: urlState().state.get().doc});
+          return;
+        }
```

Comment:
I wondered if there could be a case where an anonymous user would have access to this menu item, and there is: when they are shared a workspace or an organisation.

In such a case, the workflow is relevant: they are invited to login or to logon. So good catch, that's perfect! :ok_hand:

Reply:
Can you point me to a situation when this might happen?